diff --git a/arch/arm/mach-sunxi/Kconfig b/arch/arm/mach-sunxi/Kconfig
index 621bdfa..e4a7909 100644
--- a/arch/arm/mach-sunxi/Kconfig
+++ b/arch/arm/mach-sunxi/Kconfig
@@ -335,6 +335,12 @@ config TARGET_A20_OLINUXINO
 	---help---
 	Use board autodetection feature on A20-OLinuXino boards
 
+config TARGET_GETEK_SOM204_A20
+	bool "GETEK SOM204 A20 board"
+	select I2C1_ENABLE
+	---help---
+	Use board autodetection feature on GETEK boards
+
 endchoice
 endif
 
@@ -606,6 +612,10 @@ if TARGET_A20_OLINUXINO
 	source "board/olimex/a20_olinuxino/Kconfig"
 endif
 
+if TARGET_GETEK_SOM204_A20
+	source "board/getek/getek_som204_a20/Kconfig"
+endif
+
 if TARGET_A33_OLINUXINO
 	source "board/olimex/a33_olinuxino/Kconfig"
 endif
@@ -635,7 +645,7 @@ config OLD_SUNXI_KERNEL_COMPAT
 	Set this to enable various workarounds for old kernels, this results in
 	sub-optimal settings for newer kernels, only enable if needed.
 
-if !TARGET_A20_OLINUXINO && !TARGET_A33_OLINUXINO && !TARGET_A64_OLINUXINO
+if !TARGET_A20_OLINUXINO && !TARGET_A33_OLINUXINO && !TARGET_A64_OLINUXINO && !TARGET_GETEK_SOM204_A20
 config MACPWR
 	string "MAC power pin"
 	default ""
@@ -707,7 +717,7 @@ config INITIAL_USB_SCAN_DELAY
 	option to a non 0 value to add an extra delay before the first usb
 	bus scan.
 
-if !TARGET_A20_OLINUXINO && !TARGET_A33_OLINUXINO && !TARGET_A64_OLINUXINO
+if !TARGET_A20_OLINUXINO && !TARGET_A33_OLINUXINO && !TARGET_A64_OLINUXINO && !TARGET_GETEK_SOM204_A20
 config USB0_VBUS_PIN
 	string "Vbus enable pin for usb0 (otg)"
 	default ""
@@ -1005,6 +1015,10 @@ config VIDEO_LCD_PANEL_OLINUXINO
 	bool "Olimex LCD-OLinuXino LCD panel"
 	select VIDEO_LCD_IF_PARALLEL
 
+config VIDEO_LCD_PANEL_GETEK
+	bool "Getek LCD-GETEK LCD panel"
+	select VIDEO_LCD_IF_PARALLEL
+
 config VIDEO_LCD_PANEL_MIPI_4_LANE_513_MBPS_VIA_SSD2828
 	bool "MIPI 4-lane, 513Mbps LCD panel via SSD2828 bridge chip"
 	select VIDEO_LCD_SSD2828
@@ -1038,7 +1052,7 @@ config VIDEO_LCD_TL059WV5C0
 
 endchoice
 
-if !TARGET_A20_OLINUXINO && !TARGET_A33_OLINUXINO
+if !TARGET_A20_OLINUXINO && !TARGET_A33_OLINUXINO && !TARGET_GETEK_SOM204_A20
 config SATAPWR
 	string "SATA power pin"
 	default ""
